src/fides/core/dataset.py

```diff
@@ -409,10 +414,23 @@ def get_snowflake_schemas(
             metadata[schema] = {}
             table_cursor = engine.execute(text(f'SHOW TABLES IN "{schema}"'))
             db_tables = [row[1] for row in table_cursor]
-            for table in db_tables:
-                column_cursor = engine.execute(
-                    text(f'SHOW COLUMNS IN "{schema}"."{table}"')
-                )
-                columns = [row[2] for row in column_cursor]
+            number_of_threads = 8 if len(db_tables) > 250 else 4
```

Comment:
i think it'd be really nice to make this config-driven in some way, if that's not too crazy of a thing to do. it feels like something we'd potentially want to fine tune "on the fly" - either for pure testing or just depending on various different environmental factors. could end up saving a lot of iteration overhead, where we could simply tweak in a deployed env, rather than having to cut new releases to iterate on the values we want here.

curious to hear what you think though - maybe this would less helpful than i'm imagining!

Reply:
It felt like a bit of a pickle, I thought about it and strayed away a bit due to wanting to manage it without adding further config variables. I didn't like the idea of having to restart/re-deploy to generate a database (likely just one time and only for Snowflake). If these values didn't work, then the next step may be to up the timeout of the API or Generate the values in a separate thread entirely (similar to how we Classify today).

These aren't terribly heavy operations really as much as tedious so I did feel hesitant to add much extra, especially as it feels like there may be something better overall in a minor rewrite or other change specific to handling Snowflake

Reply:
got it, that seems like a totally fair trade off if you don't imagine needing to fine tune the number of threads!